assets/js/modules/adsense/components/common/AdBlockingRecoveryToggle.test.js

```
@@ -0,0 +1,111 @@
+	};
+
+	it.each( [
+		[ 'the Ad blocker detection feature flag is not enabled', false ],
+		[ 'Ad blocker recovery setup status is empty', true ],
+	] )(
+		'should not render the Ad blocking recovery toggle when %s',
+		( testName, adBlockerDetectionEnabled ) => {
+			const { container } = render( <AdBlockingRecoveryToggle />, {
+				setupRegistry: ( registry ) => {
+					if ( adBlockerDetectionEnabled ) {
+						enabledFeatures.add( 'adBlockerDetection' );
+					}
+					provideModules( registry, [
+						{
+							slug: 'adsense',
+							active: true,
+							connected: true,
+						},
+					] );
+					registry
+						.dispatch( MODULES_ADSENSE )
+						.receiveGetSettings( validSettings );
+				},
+			} );
+
+			expect(
+				container.querySelector(
+					'.googlesitekit-settings-module__ad-blocking-recovery-toggles'
+				)
+			).toBeNull();
+
+			expect( container.textContent ).not.toContain(
+				'Ad blocking recovery'
+			);
+		}
+	);
+
+	it( 'should render the Ad blocking recovery toggle when the conditions are met', () => {
+		const { container } = render( <AdBlockingRecoveryToggle />, {
+			setupRegistry: ( registry ) => {
+				enabledFeatures.add( 'adBlockerDetection' );
+				provideModules( registry, [
+					{
+						slug: 'adsense',
+						active: true,
+						connected: true,
+					},
+				] );
+				registry.dispatch( MODULES_ADSENSE ).receiveGetSettings( {
+					...validSettings,
+					adBlockingRecoverySetupStatus:
+						AD_BLOCKING_RECOVERY_SETUP_STATUS_TAG_PLACED,
+					useAdBlockerDetectionSnippet: true,
+					useAdBlockerDetectionErrorSnippet: true,
+				} );
+			},
+		} );
+
+		expect(
+			container.querySelector(
+				'.googlesitekit-settings-module__ad-blocking-recovery-toggles'
+			)
+		).not.toBeNull();
+
+		expect( container.textContent ).toContain( 'Ad blocking recovery' );
+		expect(
+			container.querySelector(
+				'.googlesitekit-settings-module__meta-item'
+			)
+		).toHaveTextContent( 'Place ad blocking recovery tag' );
+	} );
```

Please add couple more test here similar to the stories. Tests where it asserts\ that disabling Ad Blocking Recovery tag hides the error protection toggle, the behavior of notice on toggle change and on mount.